src/dmd/typesem.d

```diff
@@ -2387,8 +2387,12 @@ Expression getProperty(Type t, Scope* scope_, const ref Loc loc, Identifier iden
                     error(loc, "no property `%s` for type `%s`", ident.toChars(), mt.toPrettyChars(true));
                     if (auto dsym = mt.toDsymbol(scope_))
                         if (auto sym = dsym.isAggregateDeclaration())
+                        {
                             if (auto fd = search_function(sym, Id.opDispatch))
                                 errorSupplemental(loc, "potentially malformed `opDispatch`. Use an explicit instantiation to get a better error message");
+                            else if (!sym.members)
+                                errorSupplemental(loc, "`%s` `%s` is forward declared and does not have a body.", sym.kind, mt.toPrettyChars(true));
```

Comment:
D does not support forward declarations, this should probably refer to an opaque type.
Also, can this error message be triggered for an actual forward reference error?

Reply:
> D does not support forward declarations, this should probably refer to an opaque type.

changed.

> Also, can this error message be triggered for an actual forward reference error?

Sorry, I don't understand what you mean.

Reply:
`symtab` might be `null` if this error occurs early into in the semantic of `exception`. But i think it's currently impossible to write code that could trigger such scenarios. (e.g. base classes/interface for opaque classes are rejected by the parser)

Reply:
Typically, forward reference errors are issued before `noMember` is called.